### ISSUE TITLE
Renaming Nested Node Changes its Contents

### ISSUE THREAD

**Issue by LordMcMutton** [imjp94/gd-YAFSM#82]
Godot 4.2.2

This is... strange.

I made a State Node called InAir, then went inside and set up a nested state machine.

Later, I changed its name to Jump, then did more work on it.

But then, I changed its name back to InAir, and it turned the contents _back_ into what they were when it was called InAir previously, replacing the work I did. Changing it back to Jump restores that setup.

I can put the State Machine Resource file somewhere, though I can't actually upload it here.

**Comment by LordMcMutton** [comment]
Right, immediately after making this, the behavior stopped- now it's just a mangled version of the Jump edit, with transitions going into nothing:

![image](https://github.com/user-attachments/assets/8e66deb0-3018-4675-ace1-029e9723883d)

**Comment by imjp94** [comment]
Hmmm...
I encountered this bug before when renaming a nested node, it is pretty random and I have no idea how to reproduce it as well.
You can upload to github by renaming the `.tres` extension to `.txt`

**Comment by LordMcMutton** [comment]
[PlayerStates - Copy.txt](https://github.com/user-attachments/files/16575743/PlayerStates.-.Copy.txt)

It's really weird! 

Here's the file- it's a copy of the Resource before the second post happened. Hopefully it can be helpful in some way.
